diff --git a/Command/DumpCommand.php b/Command/DumpCommand.php
index 2d543cd..6fa90f7 100644
--- a/Command/DumpCommand.php
+++ b/Command/DumpCommand.php
@@ -161,8 +161,11 @@
         // start by dumping the main asset
         $this->doDump($asset, $output);
 
-        // dump each leaf if debug
-        if (isset($formula[2]['debug']) ? $formula[2]['debug'] : $this->am->isDebug()) {
+        $debug = isset($formula[2]['debug']) ? $formula[2]['debug'] : $this->am->isDebug();
+        $combine = isset($formula[2]['combine']) ? $formula[2]['combine'] : !$debug;
+
+        // dump each leaf if no combine
+        if (!$combine) {
             foreach ($asset as $leaf) {
                 $this->doDump($leaf, $output);
             }
